Set TREESITTER_LIBDIR

mjambon opened the pull request:
These two other branches should be merged at the same time:

* https://github.com/returntocorp/ocaml-tree-sitter/pull/74
* https://github.com/returntocorp/ocaml-tree-sitter-lang/pull/1

The main highlight is that the semgrep-core build now uses a base image that comes with the right version of ocaml and most external opam packages pre-installed.

I had to fix the mess that GHA creates during the git checkout (or container launch). This is taken care of by the post-checkout scripts. Its output shows what's being done.

Commit:
Point to updated submodules

Commit:
Set TREESITTER_LIBDIR as obtained from pkg-config.

Commit:
Call 'make' instead of 'dune build' for the opam install of
semgrep-core.

Commit:
Build semgrep-core with simpler 'make' command.

Commit:
Update ocaml-tree-sitter

Commit:
Update ocaml-tree-sitter submodule to make it set TREESITTER_LIBDIR.

Commit:
Fix chicken-and-egg problems

aryx reviewed (approved):
sounds good.
Not sure why the CI failure, but would be good to fix that first :)

Commit:
Update ocaml-tree-sitter

Commit:
Trying out new base image

Commit:
Remove reference to 'opam' user account.

Commit:
Debugging

Commit:
Debugging

Commit:
Debugging

Commit:
Troubleshoot GHA

Commit:
Attempt a fix to the new home created by GHA

Commit:
Try different order of steps in GHA

Commit:
Edit step name in an attempt to convince GHA to use the new flow.

Commit:
Another attempt at fixing GHA

Commit:
Try to trigger a workflow update

Commit:
Fix magic number in bash script

Commit:
Update ocaml-tree-sitter

Commit:
Update ocaml-tree-sitter to a working version. Fix opam warning, typos.

Commit:
Fix install script (?)

Commit:
Rename fix-home -> post-checkout

Commit:
Use new base image for all semgrep-core builds

Commit:
Merge remote-tracking branch 'origin/develop' into treesitter-lib

Commit:
We need to chmod /__w after all.

Commit:
Create version.ml from its own script. This script should now fail
properly if the version can't be obtained. Additionally, this attempts a
fix for the git command that fails to produce a version string in some
environments.

Commit:
Try to fix the osx build, which uses 'make setup'

Commit:
Merge remote-tracking branch 'origin/develop' into treesitter-lib
